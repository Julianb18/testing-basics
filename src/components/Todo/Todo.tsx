

interface ITodoProps {
  id: number;
  title: string;
  completed: boolean;
}

export const Todo = (props: { todo: ITodoProps }) => {
  const { id, title, completed } = props.todo;
  // const h1 = <h1>{title}</h1>;
  // const text = completed ? (
  //   <h1 style={{ textDecoration: "line-through" }}>{h1}</h1>
  // ) : (
  //   h1
  // );

  return (
    <div data-testid={id}>
      <h1 className={`${completed ? "line-through" : null}`}>{title}</h1>
    </div>
  );
};
