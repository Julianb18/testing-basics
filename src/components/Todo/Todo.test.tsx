import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Todo } from "./Todo";

afterEach(() => {
  cleanup();
});

describe("Todo", () => {
  test("should render incomplete todo", () => {
    const todo = { id: 1, title: "wash dishes", completed: false };
    render(<Todo todo={todo} />);
    // const todoElement = screen.getByTestId(1);
    const todoElement = screen.getByText("wash dishes");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("wash dishes");
    expect(todoElement).not.toHaveClass("line-through");
  });

  test("should render completed todo", () => {
    const todo = { id: 2, title: "eat", completed: true };
    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId(2);
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("eat");
    // expect(todoElement).toHaveClass("line-through");
  });
});

// snapShots
// dont need these
test("matches snapshot", () => {
  const todo = { id: 2, title: "eat", completed: true };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  //   console.log(tree);
  expect(tree).toMatchSnapshot();
});

// jest mock
