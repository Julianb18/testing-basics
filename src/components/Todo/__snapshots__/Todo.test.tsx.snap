// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid={2}
>
  <h1
    className="line-through"
  >
    eat
  </h1>
</div>
`;
